core/src/main/scala/com/stripe/agate/tensor/OnnxNumber.scala

```diff
@@ -541,14 +534,20 @@ object OnnxNumber {
       .updated(DataType.Float32, OnnxNumber.Float32)
       .updated(DataType.Float64, OnnxNumber.Float64)
 
-  def cast(from: DataType, to: DataType): from.Elem => to.Elem = {
-    val onF = forDataType(from)
-    val onT = forDataType(to)
-
-    if (onT.isInstanceOf[OnnxIntegral[_]]) { (a: from.Elem) =>
-      onT.fromLong(onF.toLong(a))
-    } else { (a: from.Elem) =>
-      onT.fromDouble(onF.toDouble(a))
+  def cast(from: DataType, to: DataType): Option[from.Elem => to.Elem] =
```

Comment:
This feels a little more like a `Try` than an `Option` here to me, but I don't feel super strongly about it.

Reply:
Yeah, I thought about it a bit too, but in all the places we're using it, we are just treating it as an option. Like: "give me this function if it exists, otherwise I'll do something else".